engine/Default/bank_report.php

```diff
@@ -16,23 +16,15 @@
 if (!$db->getNumRows()) {
 	create_error('Your alliance has no recorded transactions.');
 }
+$trans = array();
 while ($db->nextRecord()) {
-	if ($db->getField('transaction') == 'Payment') {
-		if (!$db->getField('exempt')) {
-			$trans[$db->getInt('payee_id')][WITHDRAW] += $db->getInt('amount');
-		}
-		else {
-			$trans[0][WITHDRAW] += $db->getField('amount');
-		}
-	}
-	else {
-		if (!$db->getField('exempt')) {
-			$trans[$db->getInt('payee_id')][DEPOSIT] += $db->getInt('amount');
-		}
-		else {
-			$trans[0][DEPOSIT] += $db->getInt('amount');
-		}
+	$transType = ($db->getField('transaction') == 'Payment') ? WITHDRAW : DEPOSIT;
+	$payeeId = ($db->getField('exempt')) ? 0 : $db->getInt('payee_id');
```

Comment:
Should this be `getBoolean` instead of `getField`?

Reply:
For some reason the `exempt` field is a `tinyint(1)` instead of a bool. But if `getBoolean` would correctly convert it (even if the value is not 1/0 or true/false), then it would certainly make the code clearer.

Reply:
Oh in that case it won't :(